test/unittests/parser_test.cpp

```diff
@@ -781,15 +781,15 @@ TEST(parser, start_invalid_index)
                      make_section(3, func_section) + make_section(8, start_section) +
                      make_section(10, code_section);
 
-    EXPECT_THROW_MESSAGE(parse(bin), parser_error, "invalid start function index");
```

Comment:
these tests should better be moved to `validation_test.cpp` now

Reply:
Hm, I did that quite a few times. Perhaps we should merge these small changes as is and then do a sweep of moving things across?

Reply:
Fine

Reply:
Let's coordinate with moving bigger code pieces.